wicker/core/storage.py

```diff
@@ -215,64 +215,115 @@ class S3PathFactory:
+
+        Args:
+            dataset_id (DatasetID): Id of the dataset
+            prefix (Optional[str], optional): Optional prefix to remove from file paths. Defaults to None.
+
+        Returns:
+            str: path to assets folder
+        """
         full_path = os.path.join(
             self.root_path,
             dataset_id.name,
             dataset_id.version,
             "assets",
         )
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_dataset_partition_metadata_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+    def _get_dataset_partition_metadata_path(
+        self, data_partition: DatasetPartition, prefix: Optional[str] = None
+    ) -> str:
+        """Get partition metadata path in wicker known file structure.
+
+        Args:
+            data_partition (DatasetPartition): data partition to use for path
+            prefix (Optional[str], optional): Optional prefix to remove from file paths. Defaults to None.
+
+        Returns:
+            str: Path to partition metadata json file.
+        """
         full_path = os.path.join(
-            self.get_dataset_partition_path(data_partition),
+            self._get_dataset_partition_path(data_partition),
             "_l5ml_dataset_partition_metadata.json",
         )
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_dataset_partition_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+    def _get_dataset_partition_path(self, data_partition: DatasetPartition, prefix: Optional[str] = None) -> str:
+        """Get the dataset partition parquet path.
+
+        Private gettr handling logic of pathing centrally.
+
+        Args:
+            data_partition (DatasetPartition): DatasetPartition to use for pathing
+            prefix (Optional[str], optional): Optional prefix to remove from file paths. Defaults to None.
+
+        Returns:
+            str: Path to partition parquet file
+        """
         full_path = os.path.join(
             self.root_path,
             data_partition.dataset_id.name,
             data_partition.dataset_id.version,
             f"{data_partition.partition}.parquet",
         )
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_dataset_schema_path(self, dataset_id: DatasetID, s3_prefix: bool = True) -> str:
+    def _get_dataset_schema_path(self, dataset_id: DatasetID, prefix: Optional[str] = None) -> str:
+        """Get the dataset schema path.
+
+        Private gettr handling pathing logic to avro_schema json file.
```

Comment:
Nit: typo/spelling: `gettr` -> `getter`.

```suggestion
        Private getter handling pathing logic to avro_schema json file.
```